File: 2129-number-of-pairs-of-interchangeable-rectangles/solution.cpp
class Solution {
public:
    long long interchangeableRectangles(vector<vector<int>>& rect) {
        long long pc =0;
        unordered_map<double,int> m;
        for(auto ele:rect){
            double ratio  = (double) ele[0] / ele[1];
            pc += m[ratio];
            m[ratio] += 1;
            cout<<m[ratio]<<" "<<pc<<"\n";
        }
        
        return pc;
    }
};

File: 2129-number-of-pairs-of-interchangeable-rectangles/solution.java
class Solution {
    public long interchangeableRectangles(int[][] rect) {
        long cnt = 0;
        int n = rect.length;
        Map<Double,Integer> m = new HashMap<>();
        for(int[] i : rect){
            double ratio = (double) i[0] / i[1];
            cnt += m.getOrDefault(ratio,0);
            m.put(ratio,m.getOrDefault(ratio,0)+1);
        }
        return cnt;
    }
}
